File: solutions/vbnet/nth-prime/1/NthPrime.vb
Imports System

Public Module NthPrime
  
    Public Function Prime(ByVal number As Integer) As Integer

        If number < 1 Then Throw New ArgumentOutOfRangeException

        Dim isprime = Function(n As Integer) As Boolean
                          If n < 2 Then Return False
                          If n = 2 Then Return True
                          If n Mod 2 = 0 Then Return False

                          Dim limite As Integer = CInt(Math.Sqrt(n))
                          For i As Integer = 3 To limite Step 2
                              If n Mod i = 0 Then Return False
                          Next
                          Return True
                      End Function

        Dim count As Integer = 0
        Dim num As Integer = 1

        While count < number
            num += 1
            If isprime(num) Then
                count += 1
            End If
        End While

        Return num

    End Function

End Module

File: solutions/euphoria/nth-prime/1/nth-prime.ex
public function prime(integer number)
    integer count
    integer candidate
    integer is_prime

    if number < 1 then
        return -1
    end if

    if number = 1 then
        return 2
    end if

    count = 1
    candidate = 1

    while count < number do
        candidate += 2
        is_prime = 1

        for d = 3 to floor(sqrt(candidate)) by 2 do
            if remainder(candidate, d) = 0 then
                is_prime = 0
                exit
            end if
        end for

        if is_prime then
            count += 1
        end if
    end while

    return candidate
end function